dev/bots/codesign.dart

```diff
@@ -37,10 +37,46 @@ List<String> findBinaryPaths() {
   return allFiles.where(isBinary).toList();
 }
 
+/// Given the path to a stamp file, read the contents.
+///
+/// Will throw if the file doesn't exist.
+String readStamp(String filePath) {
+  final File file = File(filePath);
+  if (!file.existsSync()) {
+    throw 'Error! Stamp file $filePath does not exist!';
+  }
+  return file.readAsStringSync().trim();
+}
+
+/// Return whether or not the flutter cache is up to date.
+bool checkCacheIsCurrent() {
+  try {
+    final String dartSdkStamp = readStamp(path.join(cacheDirectory, 'engine-dart-sdk.stamp'));
+    final String engineVersion = readStamp(path.join(repoRoot, 'bin', 'internal', 'engine.version'));
+    return dartSdkStamp == engineVersion;
+  } catch (e) {
+    print(e);
+    return false;
+  }
+}
+
 void main() {
   final List<String> failures = <String>[];
 
-  for (final String binaryPath in findBinaryPaths()) {
+  if (!Platform.isMacOS) {
+    print('Warning! This script only works on macOS, as it requires Xcode.');
```

Comment:
uber-nit: on CI, this might read strangely if you thought it was running on macOS, and it might be helpful to just print out `Platform.operatingSystem` in here.

Ideally we'd also want to make sure the binaries we need (e.g. `codesign`) are on the path, but it's probably fine to skip that for this script.

Reply:
updated error message.